### PR TITLE
Update pu/pu to support the new settings pull command + new deployment file

### PR DESCRIPTION
**Pull request by glena**
# Description

- Add new deployment settings pull command: this command will pull the deployment settings from pulumi cloud and generate the new deployment file. For now it will be hidden until we have completed the whole feature.
- Add support for the new deployment file, this will contain all the deployment related information (for now just the settings), example:

```
settings:
  -executorContext: {}
    sourceContext:
      git:
        branch: main
        repoDir: .
    gitHub:
      repository: glena/test-action
      deployCommits: true
      previewPullRequests: false
    operationContext:
      preRunCommands: []
      operation: ""
      environmentVariables: {}
      options:
        skipInstallDependencies: false
        skipIntermediateDeployments: true
        shell: ""
        deleteAfterDestroy: false
        remediateIfDriftDetected: false
    agentPoolID: ...
```

Fixes https://github.com/pulumi/pulumi-service/issues/20306

## Checklist

- [ ] I have run `make tidy` to update any new dependencies
- [ ] I have run `make lint` to verify my code passes the lint check
  - [ ] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/cmd/pulumi/deployment.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+			deploymentSettings, err := currentBe.GetStackDeploymentSettings(ctx, s)
+			if err != nil {
+				return err
+			}
+
+			newStackDeployment := &workspace.ProjectStackDeployment{
+				DeploymentSettings: *deploymentSettings,
+			}
+
+			err = saveProjectStackDeployment(newStackDeployment, s)
```
Comment (glena) [inline pkg/cmd/pulumi/deployment.go:130]: TODO: use marshalling logic similar to the project file

File: pkg/cmd/pulumi/deployment.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	cmd.AddCommand(newDeploymentSettingsCmd())
+
+	return cmd
+}
+
+func newDeploymentSettingsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "settings",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Manages the stack's deployment settings",
```
Comment (lunaris) [inline pkg/cmd/pulumi/deployment.go:74]: In general I'd try and keep these all active like the one you have above, so `Manage stack deployment settings` for instance.

File: sdk/go/common/workspace/paths.go [hunk 3]
```diff
@@ -60,6 +60,8 @@ const (
 
 	// ProjectFile is the base name of a project file.
 	ProjectFile = "Pulumi"
+	// DeploymentSuffix base suffix for the deployment file
```
Comment (lunaris) [inline sdk/go/common/workspace/paths.go:63]:
```suggestion
	// DeploymentSuffix is the base suffix for deployment settings files (e.g. "Pulumi.<stack>.deploy.yaml").
```

File: sdk/go/common/workspace/paths_test.go [hunk 4]
```diff
@@ -142,3 +142,29 @@ func TestDetectProjectUnreadableParent(t *testing.T) {
 	_, _, err = DetectProjectAndPath()
 	assert.ErrorIs(t, err, ErrProjectNotFound)
 }
+
+//nolint:paralleltest // Theses test use and change the current working directory
```
Comment (lunaris) [inline sdk/go/common/workspace/paths_test.go:146]:
```suggestion
//nolint:paralleltest // These tests use and change the current working directory
```

File: sdk/go/common/workspace/paths_test.go [hunk 5]
```diff
@@ -142,3 +142,29 @@ func TestDetectProjectUnreadableParent(t *testing.T) {
...
+func TestDetectProjectStackDeploymentPath(t *testing.T) {
+	tmpDir := mkTempDir(t)
+	cwd, err := os.Getwd()
+	assert.NoError(t, err)
+	defer func() { err := os.Chdir(cwd); assert.NoError(t, err) }()
+	err = os.Chdir(tmpDir)
+	assert.NoError(t, err)
+
+	yamlPath := filepath.Join(tmpDir, "Pulumi.yaml")
+	yamlContents := "name: some_project\ndescription: Some project\nruntime: nodejs\n"
```
Comment (lunaris) [inline sdk/go/common/workspace/paths_test.go:156]: Maybe it's just me but I feel like a multi-line string could be more readable here? So e.g.

```go
yamlContents := `name: some_project
description: Some project
runtime: nodejs`
```

**Review by lunaris — APPROVED** [review]

**Pull request merged**